src/azure-cli/azure/cli/command_modules/resource/custom.py

```diff
@@ -1007,6 +1016,10 @@ def _update_provider(cli_ctx, namespace, registering, wait):
             rp_info = rcf.providers.get(namespace)
             if rp_info.registration_state == target_state:
                 break
+        if is_rpaas and registering:
+            # call accept term API
+            from azure.cli.core.util import send_raw_request
+            send_raw_request(cli_ctx, 'put', RPAAS_APIS[namespace.lower()], body=json.dumps({"properties": {"accepted": True}}))
```

Comment:
Do we need to consider checking the return results of the requests for REST? Such as checking the results of the operation, whether there are exceptions, and so on

Reply:
If the result is not successful, an exception is thrown.
https://github.com/Azure/azure-cli/blob/a3b384eba388bc31003967769ede09a91c162571/src/azure-cli-core/azure/cli/core/util.py#L914-L918
This will leave the exception to be handled by the exception_handler.

Reply:
OK